### PR TITLE
[CR] Add Build System support for POWER8/VSX

### PR DESCRIPTION
**Pull request by rafaeldelucena**
## Description

Add support for Power8 and VSX instructions in order to resolve the issue #1 

## Checklist:

- [x] Detect PPC toolchain
- [x] Enable VSX flags when available
- [x] Check at runtime if Linux has VSX feature enabled
- [x] Generate appropriate headers for PPC port

### CODE REVIEW

**Review by lu-zero — COMMENTED** [review]
Overall I'm ambivalent, if you prefer having it short so be it =)

File: build/make/configure.sh [hunk 1]
```diff
@@ -697,6 +697,9 @@ process_common_toolchain() {
       *sparc*)
         tgt_isa=sparc
         ;;
+      p*pc*)
```
Comment (lu-zero) [inline build/make/configure.sh]: This is dangerous, better do

```
ppc*)
power*)
```

File: build/make/configure.sh [hunk 2]
```diff
@@ -1147,6 +1153,24 @@ EOF
...
+            RTCD_OPTIONS="${RTCD_OPTIONS}--disable-${ext} "
+            disable_feature $ext
+          fi
+        elif disabled $ext; then
+          disable_exts="yes"
+        else
+          check_gcc_machine_option $ext $ext
+        fi
+      done
+
```
Comment (lu-zero) [inline build/make/configure.sh]: nit: stray line

File: vpx_ports/ppc.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#include "vpx_config.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int ppc_simd_caps(void);
+
+// Earlier gcc compilers doesn't have support for Altivec
+#if !defined(__clang__) && defined(__GNUC__) && __GNUC__ == 4 && \
```
Comment (lu-zero) [inline vpx_ports/ppc.h:26]: VSX might require a newer version. I'd use 6 to be safe, 5.4 is known to have horrible bugs =P

File: vpx_ports/ppc_cpudetect.c [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#include <fcntl.h>
+#include <unistd.h>
+#include <asm/cputable.h>
+#include <linux/auxvec.h>
+
+int ppc_simd_caps(void) {
```
Comment (lu-zero) [inline vpx_ports/ppc_cpudetect.c]: Linux-only code...

File: vpx_ports/ppc.h (hunk 3 above)
Comment (rafaeldelucena) [reply vpx_ports/ppc.h:26]: Do you mean the gcc version 4.6 is enough?

Comment (lu-zero) [reply vpx_ports/ppc.h:26]: no, that gcc 6.2 could be the least version non-broken.

Comment (rafaeldelucena) [reply vpx_ports/ppc.h:26]: Weird, because I'm testing the build on ppc64le with version 4.9

File: vpx_ports/ppc_cpudetect.c (hunk 4 above)
Comment (rafaeldelucena) [reply vpx_ports/ppc_cpudetect.c]: Defines are fine here? or you want that build system detects and add the code for the correct OS?

File: vpx_ports/ppc.h (hunk 3 above)
Comment (lu-zero) [reply vpx_ports/ppc.h:26]: I got bitten recently by this bug, possibly it is version 5.4-only, but we never know =)

File: vpx_ports/ppc_cpudetect.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#if defined(__linux__)
+#include <fcntl.h>
+#include <unistd.h>
+#include <asm/cputable.h>
+#include <linux/auxvec.h>
+#endif
+
+int ppc_simd_caps(void) {
+	int result = 0;
+#if defined(__linux__)
```
Comment (lu-zero) [inline vpx_ports/ppc_cpudetect.c]: Seems matching what happens in other architectures, thank you!

You should also consider `CONFIG_RUNTIME_CPU_DETECT` from what I'm seeing.

**Comment by lu-zero** [comment]
The set looks ok, could you please squash it in a single commit?

**Comment by rafaeldelucena** [comment]
What are the plans to integrate with upstream? Will you create an unique PR or for each feature?

**Comment by lu-zero** [comment]
Probably I'd send a first set with your patch and some initial code using it and then trickle in further improvements, no need to push a single monster set.

**Comment by lu-zero** [comment]
Did you sign the CLA? I was waiting for your ok to send over something and see how upstream takes it :)

**Comment by rafaeldelucena** [comment]
Hello,

I already did, it's all right :D

**Pull request merged**